DSCResources/cTentacleAgent/cTentacleAgent.psm1

```diff
@@ -244,11 +247,12 @@ function Invoke-AndAssert {
 # After the Tentacle is registered with Octopus, Tentacle listens on a TCP port, and Octopus connects to it. The Octopus server
 # needs to know the public IP address to use to connect to this Tentacle instance. Is there a way in Windows Azure in which we can 
 # know the public IP/host name of the current machine?
-function Get-MyPublicIPAddress
+function Get-MyPublicIPAddress([string]$RegisteredNic)
 {
-    Write-Verbose "Getting public IP address"
-    $downloader = new-object System.Net.WebClient
-    $ip = $downloader.DownloadString("http://icanhazip.com").Trim()
+    Write-Verbose "Getting IP address of $($RegisteredNic) NIC"
+    $ip = Get-NetIPAddress -InterfaceAlias $RegisteredNic -AddressFamily IPv4 | select -exp IPAddress
```

Comment:
I can think of scenarios where this would not actually yield the desired public IP address. For instance a server behind a NAT (think VM in dedicated) with the Octopus server being unable to reach the tentacle via private IP (e.g. Octopus server located on customer's premise without VPN). I remember struggling with the same problem when porting this module for my customer and not entirely sure how to fix. Maybe add a switch that indicates whether we want NATed or private/adapter address?